synapse/media/media_storage.py

```diff
@@ -95,10 +97,12 @@ async def store_file(self, source: IO, file_info: FileInfo) -> str:
 
         return fname
 
+    @trace
     async def write_to_file(self, source: IO, output: IO) -> None:
         """Asynchronously write the `source` to `output`."""
         await defer_to_thread(self.reactor, _write_file_synchronously, source, output)
 
+    @trace
     @contextlib.contextmanager
     def store_into_file(
```

Comment:
Only questionable `@trace` is here. I think stacking this would work fine but should just sanity check that file upload works (and hope that we would have tests in CI that would cover it in any case)

Reply:
Seems to work fine :white_check_mark: 

Example trace in the description is with these changes.